### PR TITLE
Table block: Fix focus loss in between row/column insertions.

### LINKED ISSUE #23103: Table Block - Adding More than One or Multiple Rows
**Problem:**
After adding a table row with a mouse, the focus is removed from the table. Attempting to add a new row immediately, the option to add a new row disappears, as the table has lost focus in the editor.

**Describe the solution you'd like**

- After adding a new row to a table, the selection in the editing screen should be within that new row. That way, adding a second new row, it will work.
- An alternative is to allow the user to enter a number of rows they wish to add.

**Describe alternatives you've considered**

- I did a brief search for table row add shortcuts for Gutenberg, I did not immediately see any.
- I tried using a return to add a new row, however this only adds a content return within the table cell.

gutenburg-table-row

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description

Fixes #23103

This PR just allows to be able to add more rows/columns, by enabling the corresponding buttons, after having added a row/column.

To reproduce what happens now:
- Go to a post/page
- Add a `Table` block ( ex 2x2 ) and insert some text in a cell
- Go to the `Toolbar` and click `Add a row Before/After`
- Go to the `Toolbar` again and in the Dropdown menu for adding rows you'll see that everything is disabled

Other enhancements on `Table` block can be handled separately, such as adding keyboard support for some actions. 

It needs to be revised when programmatically focus to RichText is handled, as mentioned here #9740.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/table/edit.js [hunk 1]
```diff
@@ -291,14 +291,21 @@ export class TableEdit extends Component {
...
+				rowIndex: newRowIndex,
 			} )
 		);
+		// this just allows to be able to add more rows, by enabling the corresponding
+		// buttons. It needs to be revised when programmatically focus to RichText is handled
+		this.createOnFocus( {
+			sectionName,
+			rowIndex: newRowIndex,
+			columnIndex: 0,
+		} );
```
Comment (mcsf) [inline packages/block-library/src/table/edit.js:308]: By definition, `createOnFocus` just returns a new function but doesn't do anything with it. So I'm fairly sure that nothing is happening here. :)

File: packages/block-library/src/table/edit.js [hunk 2]
```diff
@@ -346,13 +353,19 @@ export class TableEdit extends Component {
...
-				columnIndex: columnIndex + delta,
+				columnIndex: newColumnIndex,
 			} )
 		);
+		// this just allows to be able to add more rows, by enabling the corresponding
+		// buttons. It needs to be revised when programmatically focus to RichText is handled
+		this.createOnFocus( {
+			rowIndex: 0,
+			columnIndex: newColumnIndex,
+		} );
```
Comment (mcsf) [inline packages/block-library/src/table/edit.js:368]: Same observation as for rows.

**Review by mcsf — CHANGES_REQUESTED** [review]
I think some clear testing instructions for this pull request would be very helpful, since the issue isn't obvious to reproduce consistently.

I'm glad you link to #9740, I think that's useful context to have. That said, my first impression is that we can best address the current Table issue by finding what — if anything — we can do at the DOM-event level to mitigate the issue.

File: packages/block-library/src/table/edit.js (hunk 1 above)
Comment (ntsekouras) [reply packages/block-library/src/table/edit.js]: Thanks Miguel. I couldn't understand why I left it that way... Maybe when I was trying to find how to focus on RichText..?

I changed it to just change the selected cell in state.

File: packages/block-library/src/table/edit.js [hunk 3]
```diff
@@ -291,14 +291,24 @@ export class TableEdit extends Component {
...
-		this.setState( { selectedCell: null } );
 		setAttributes(
 			insertRow( attributes, {
 				sectionName,
-				rowIndex: rowIndex + delta,
+				rowIndex: newRowIndex,
 			} )
 		);
+		// this just allows to be able to add more rows, by enabling the corresponding
+		// buttons. It needs to be revised when programmatically focus to RichText is handled
```
Comment (mcsf) [inline packages/block-library/src/table/edit.js:303]:
```suggestion
		// Select the first cell of the new row
```

I think the above is enough. Right now we have an adequate solution; if we had something more "hacky", some context would indeed be useful. And we already have the RichText issue open for any future enhancements.

File: packages/block-library/src/table/edit.js [hunk 4]
```diff
@@ -346,13 +356,22 @@ export class TableEdit extends Component {
...
 
-		this.setState( { selectedCell: null } );
 		setAttributes(
 			insertColumn( attributes, {
-				columnIndex: columnIndex + delta,
+				columnIndex: newColumnIndex,
 			} )
 		);
+		// this just allows to be able to add more rows, by enabling the corresponding
+		// buttons. It needs to be revised when programmatically focus to RichText is handled
```
Comment (mcsf) [inline packages/block-library/src/table/edit.js:367]:
```suggestion
		// Select the first cell of the new column
```

**Review by mcsf — APPROVED** [review]

**Comment by mcsf** [comment]
Nice fix!

**Comment by cngodles** [comment]
Thank you. I'll be so glad when I can add rows easier. I have a few situations where I need to go to an existing table and add 20-30 at a time. This will cut down on clicks.

**Pull request merged**